### PR TITLE
Fix native UI save_dialog when there are no saves

### PR DESCRIPTION
**Pull request by isJuhn**
Disallow you from selecting a save to load when there are no saves. This is more in line with how the QT save dialog works and fixes fatal access violation in Ratchet & Clank 3 (and probably in more games).

### CODE REVIEW

**Comment by kd-11** [comment]
This is a hack. The check should be done elsewhere but the selection being -1 is not the problem..

**Comment by isJuhn** [comment]
Just booted it up and tested, it just displays the text "There is no saved data." in the middle of the screen. With only a back button. I'll see if I can come up with something as close to that as possible.

**Comment by isJuhn** [comment]
rpcs3_2018-01-26_22-08-32

**Comment by kd-11** [comment]
Looks kinda wierd, move the cancel button to the position occupied by the accept button

File: rpcs3/Emu/RSX/overlays.h [hunk 1]
```diff
@@ -353,6 +360,19 @@ namespace rsx
 					m_list->add_entry(new_stub);
 				}
 
+				if (!m_list->m_items.size())
+				{
+					m_no_saves_text = std::make_unique<label>();
+					m_no_saves_text->set_font("Arial", 20);
+					m_no_saves_text->set_pos(515, 345);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/overlays.h]: Set the pos and size to span the whole width of the list view and set text alignment to center. Remember not all fonts have the same metrics.

File: rpcs3/Emu/RSX/overlays.h [hunk 2]
```diff
@@ -353,6 +360,19 @@ namespace rsx
...
+					m_no_saves_text->set_text("There is no saved data.");
+					m_no_saves_text->back_color.a = 0;
+					static_cast<label*>(m_no_saves_text.get())->auto_resize();
```
Comment (kd-11) [inline rpcs3/Emu/RSX/overlays.h]: Remove this once you do the center alignment thing from above

**Comment by isJuhn** [comment]
Fixed

**Pull request merged**